[Bug] [Connector-V2] Sync TDEngine to Doris，Doesn't support TDENGINE type 'NCHAR' yet.

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

Sync TDEngine to Doris，Error：Doesn't support TDENGINE type 'NCHAR' yet.



![image](https://github.com/user-attachments/assets/4258c1b4-28b4-43da-b4e1-bab4bc3031de)



### SeaTunnel Version

SeaTunnel Version
2.3.8

### SeaTunnel Config

```conf
env {
    parallelism = 1
    job.mode = "STREAMING"
    checkpoint.interval = 60000
}

source {
    TDengine {
        url : "jdbc:TAOS-RS://192.168.1.239:6041/"
        username : "it"
        password : "123456"
        database : "TEST_cloud"
        stable : "sys_coldata"
        lower_bound : "2024-01-03 10:38:05.000"
        upper_bound : "2024-12-30 15:00:00.800"
        plugin_output = "tdengine_result"
    }

}

sink {
    Doris {
        fenodes = "192.168.1.91:8030"
        username = root
        password = ${DORIS_ROOT_PWD}
        database = TEST
        table = "${table_name}_cloud"
        sink.label-prefix = "cloud"
        sink.enable-2pc = "false"
        sink.enable-delete = "true"
        doris.config {
            format = "json"
            read_json_by_line = "true"
        }
    }
}
```


### Running Command

```shell
./bin/seatunnel.sh --config job/td-jdbc-doris --async -n td-test
```


### Error Exception

```log
Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:213)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.connectors.seatunnel.tdengine.exception.TDengineConnectorException: ErrorCode:[COMMON-07], ErrorDescription:[Unsupported data type] - Doesn't support TDENGINE type 'NCHAR' yet.
        at org.apache.seatunnel.connectors.seatunnel.tdengine.typemapper.TDengineTypeMapper.mapping(TDengineTypeMapper.java:150)
        at org.apache.seatunnel.connectors.seatunnel.tdengine.source.TDengineSource.getStableMetadata(TDengineSource.java:156)
        at org.apache.seatunnel.connectors.seatunnel.tdengine.source.TDengineSource.prepare(TDengineSource.java:93)
        at org.apache.seatunnel.engine.core.parse.JobConfigParser.parseSource(JobConfigParser.java:83)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:370)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:209)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:114)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:182)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:160)
        ... 2 more
```


### Zeta or Flink or Spark Version

Zeta

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-tdengine/src/main/java/org/apache/seatunnel/connectors/seatunnel/tdengine/typemapper/TDengineTypeMapper.java
+++ b/seatunnel-connectors-v2/connector-tdengine/src/main/java/org/apache/seatunnel/connectors/seatunnel/tdengine/typemapper/TDengineTypeMapper.java
@@ -58,6 +58,7 @@ public class TDengineTypeMapper {
 
     // -------------------------string----------------------------
     private static final String TDENGINE_CHAR = "CHAR";
+    private static final String TDENGINE_NCHAR = "NCHAR";
     private static final String TDENGINE_VARCHAR = "VARCHAR";
     private static final String TDENGINE_TINYTEXT = "TINYTEXT";
     private static final String TDENGINE_MEDIUMTEXT = "MEDIUMTEXT";
@@ -118,6 +119,7 @@ public class TDengineTypeMapper {
                 log.warn("{} will probably cause value overflow.", TDENGINE_DOUBLE_UNSIGNED);
                 return BasicType.DOUBLE_TYPE;
             case TDENGINE_CHAR:
+            case TDENGINE_NCHAR:
             case TDENGINE_TINYTEXT:
             case TDENGINE_MEDIUMTEXT:
             case TDENGINE_TEXT:

--- a/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-tdengine-e2e/src/test/java/org/apache/seatunnel/e2e/connector/tdengine/TDengineIT.java
+++ b/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-tdengine-e2e/src/test/java/org/apache/seatunnel/e2e/connector/tdengine/TDengineIT.java
@@ -109,7 +109,7 @@ public class TDengineIT extends TestSuiteBase implements TestResource {
         try (Statement stmt = connection1.createStatement()) {
             stmt.execute("CREATE DATABASE power KEEP 3650");
             stmt.execute(
-                    "CREATE STABLE power.meters (ts TIMESTAMP, current FLOAT, voltage INT, phase FLOAT, off BOOL) "
+                    "CREATE STABLE power.meters (ts TIMESTAMP, current FLOAT, voltage INT, phase FLOAT, off BOOL, nc NCHAR(10)) "
                             + "TAGS (location BINARY(64), groupId INT)");
             String sql = getSQL();
             rowCount = stmt.executeUpdate(sql);
@@ -117,7 +117,7 @@ public class TDengineIT extends TestSuiteBase implements TestResource {
         try (Statement stmt = connection2.createStatement()) {
             stmt.execute("CREATE DATABASE power2 KEEP 3650");
             stmt.execute(
-                    "CREATE STABLE power2.meters2 (ts TIMESTAMP, current FLOAT, voltage INT, phase FLOAT, off BOOL) "
+                    "CREATE STABLE power2.meters2 (ts TIMESTAMP, current FLOAT, voltage INT, phase FLOAT, off BOOL, nc NCHAR(10)) "
                             + "TAGS (location BINARY(64), groupId INT)");
         }
         return rowCount;
@@ -210,6 +210,8 @@ public class TDengineIT extends TestSuiteBase implements TestResource {
                     .append(ps[4])
                     .append(",") // off
                     .append(ps[7])
+                    .append(",") // nc
+                    .append(ps[8])
                     .append(") "); // phase
         }
         return sb.toString();
@@ -217,13 +219,13 @@ public class TDengineIT extends TestSuiteBase implements TestResource {
 
     private static List<String> getRawData() {
         return Arrays.asList(
-                "d1001,2018-10-03 14:38:05.000,10.30000,219,0.31000,'California.SanFrancisco',2,true",
-                "d1001,2018-10-03 14:38:15.000,12.60000,218,0.33000,'California.SanFrancisco',2,false",
-                "d1001,2018-10-03 14:38:16.800,12.30000,221,0.31000,'California.SanFrancisco',2,true",
-                "d1002,2018-10-03 14:38:16.650,10.30000,218,0.25000,'California.SanFrancisco',3,true",
-                "d1003,2018-10-03 14:38:05.500,11.80000,221,0.28000,'California.LosAngeles',2,true",
-                "d1003,2018-10-03 14:38:16.600,13.40000,223,0.29000,'California.LosAngeles',2,true",
-                "d1004,2018-10-03 14:38:05.000,10.80000,223,0.29000,'California.LosAngeles',3,true",
-                "d1004,2018-10-03 14:38:06.500,11.50000,221,0.35000,'California.LosAngeles',3,false");
+                "d1001,2018-10-03 14:38:05.000,10.30000,219,0.31000,'California.SanFrancisco',2,true,'nc'",
+                "d1001,2018-10-03 14:38:15.000,12.60000,218,0.33000,'California.SanFrancisco',2,false,'nc'",
+                "d1001,2018-10-03 14:38:16.800,12.30000,221,0.31000,'California.SanFrancisco',2,true,'nc'",
+                "d1002,2018-10-03 14:38:16.650,10.30000,218,0.25000,'California.SanFrancisco',3,true,'nc'",
+                "d1003,2018-10-03 14:38:05.500,11.80000,221,0.28000,'California.LosAngeles',2,true,'nc'",
+                "d1003,2018-10-03 14:38:16.600,13.40000,223,0.29000,'California.LosAngeles',2,true,'nc'",
+                "d1004,2018-10-03 14:38:05.000,10.80000,223,0.29000,'California.LosAngeles',3,true,'nc'",
+                "d1004,2018-10-03 14:38:06.500,11.50000,221,0.35000,'California.LosAngeles',3,false,'nc'");
     }
 }

